src/cloudWatchLogs/activation.ts

```diff
@@ -18,11 +18,18 @@ import { LogStreamRegistry } from './registry/logStreamRegistry'
 import { Commands } from '../shared/vscode/commands2'
 import { searchLogGroup } from './commands/searchLogGroup'
 import { changeLogSearchParams } from './changeLogSearch'
+import { JumpToStream } from './commands/searchLogGroup'
 
 export async function activate(context: vscode.ExtensionContext, configuration: Settings): Promise<void> {
     const settings = new CloudWatchLogsSettings(configuration)
     const registry = new LogStreamRegistry(settings)
 
+    const definitionProvider = new JumpToStream(registry)
+    vscode.languages.registerDefinitionProvider(
+        { language: 'log', scheme: CLOUDWATCH_LOGS_SCHEME, pattern: `**` },
```

Comment:
Struggling to find what the correct pattern here is. I thought it would be something like `[!:]?[!:]**:[!:]?[!:]**`, but that doesn't seem to work. Looking to only match strings that contain a single `:` with one or more characters on either side.

Reply:
Why do we need this pattern?
You'd use this: `.*:.+`

Reply:
This makes it so the jumpto `definitionProvider` only runs on the log search files rather than the single stream files. Right now, it just throws an error when it can't find the streamID causing the `definitionProvider` to fail on log stream files. Uri for log search takes form '{region}:{group}` whereas for stream its '{region}:{group}:{stream}`.

Reply:
I'm not getting that this pattern matches either document. Is there a step I am missing beyond replacing the pattern field with this string?

Reply:
Oh I think `pattern` is a glob pattern. So you'd use `**/.*:.+`

But I think the better question is why there's no stream ID for a log search. Shouldn't both searches have it?

Reply:
There is no streamID because the log search is combining the results from all the streams in a log group so it's not limited to a single stream.

Reply:
Also, I am still having trouble with the `pattern`. I feel like I must have something wrong. I just committed what I currently have.

Reply:
> There is no streamID because the log search is combining the results from all the streams in a log group so it's not limited to a single stream.

What's the value of this line tracking logic + definition provider if it always jumps to the same stream?

Reply:
Yes exactly.

Reply:
You won't be able to do what you're trying to do w/ glob patterns. I'd omit the pattern for now and make a backlog item. We'd probably use a different language feature to avoid this problem, like rendering ghost text somewhere by relevant events.